src/components/core/MultipleSelect.tsx

```diff
@@ -0,0 +1,284 @@
+    suggestionElement: {
+      ...element,
+      padding: '11px 24px',
+      backgroundColor: `${theme.palette.neutralMedium}`,
+      flexShrink: 0,
+      overflowX: 'auto',
+      marginBottom: `${theme.shape.spacing(1)}px`,
+      '&:hover': {
+        cursor: 'pointer',
+        backgroundColor: `${theme.palette.primary}`,
+      },
+    },
+    input: {
+      width: 275,
+      border: 'none',
+      height: 35,
+      padding: '0 15px',
+      fontSize: 16,
+      '&:focus': {
+        outline: 'none',
+      },
+      '&::placeholder': {
+        fontSize: 24,
+      },
+    },
+    disabled: {
+      color: '#616161',
+      '&:hover': {
+        cursor: 'not-allowed',
+        backgroundColor: '#fff',
+      },
+    },
+  };
+});
+
+interface MultipleSelectProps {
+  elements: Array<string>;
+  onChange: (elements: string[]) => void;
+  allowUnknownValues?: boolean;
+  defaultValue?: string[];
+  disabled?: boolean;
+  label?: string;
+  onTop?: boolean;
+  height?: number;
+}
+
+const MultipleSelect: FC<MultipleSelectProps> = ({
+  elements: _elements,
+  onChange,
+  allowUnknownValues,
+  defaultValue,
+  disabled,
+}) => {
+  const select = useRef<HTMLDivElement>(document.createElement('div'));
+  const input = useRef<HTMLInputElement>(document.createElement('input'));
+  const [elements, setElements] = useState<Array<string>>(_elements || []);
+  const [elementsNoFilter, setElementsNoFilter] = useState<Array<string>>(_elements || []);
+  const [selectedElements, setSelected] = useState(defaultValue || []);
+  const [isNoMatches, setNoMatches] = useState(false);
+  const styles = useMultipleSelectStyles();
+
+  useEffect(() => setElements(_elements), [_elements]);
+  useEffect(() => {
+    if (defaultValue && defaultValue.length > 0) {
+      setSelected(defaultValue);
+    }
+  }, [defaultValue]);
+
+  const resetInput = (): void => {
+    input.current.value = '';
+  };
+
+  const handleSelect = (e, value) => {
+    const isAlreadySelected = selectedElements.find(el => el === value);
+    if (isAlreadySelected) return;
+
+    const newSelected = [...selectedElements, value];
+    const newElements = elementsNoFilter.filter(el => el !== value);
+
+    resetInput();
+    setSelected(newSelected);
+    setElements(newElements);
+    setElementsNoFilter(newElements);
+    onChange(newSelected);
+  };
+
+  const handleRemove = (e, value) => {
+    const newSelected = selectedElements.filter(el => el !== value);
+    const newElements = [...elements, value];
+
+    setSelected(newSelected);
+    setElements(newElements);
+    setElementsNoFilter(newElements);
+    onChange(newSelected);
+  };
+
+  const handleInputChange = e => {
+    const value = e.target.value.toLowerCase();
+
+    if (allowUnknownValues && e.key === 'Enter' && value !== '') {
+      const isAlreadySelected = selectedElements.find(el => el === value);
+      if (isAlreadySelected) return;
+
+      const newElements = elementsNoFilter.filter(el => el !== value);
```

Comment:
Don't think elements should be changed when changing the input. The tags are there for quick access. I don't think we should be filtering them. Could confuse the user as to what we're filtering - tags or people?

Reply:
this line is for when a user hits enter with filled input, but yeah, I will change the filtering of the elements while typing in the input